gateway/src/apicast/policy/oidc_authentication/oidc_authentication.lua

```diff
@@ -0,0 +1,91 @@
+-- This is a oidc_authentication description.
+local lrucache = require('resty.lrucache')
+local OIDC = require('apicast.oauth.oidc')
+local oidc_discovery = require('resty.oidc.discovery')
+local http_authorization = require('resty.http_authorization')
+local resty_url = require('resty.url')
+local policy = require('apicast.policy')
+local _M = policy.new('oidc_authentication')
+
+local tostring = tostring
+
+_M.cache_size = 100
+
+function _M.init()
+  _M.cache = lrucache.new(_M.cache_size)
+end
+
+local function valid_issuer_endpoint(endpoint)
+  return resty_url.parse(endpoint) and endpoint
+end
+
+local new = _M.new
+--- Initialize a oidc_authentication
+-- @tparam[opt] table config Policy configuration.
+function _M.new(config)
+  local self = new(config)
+
+  self.issuer_endpoint = valid_issuer_endpoint(config and config.issuer_endpoint)
+  self.discovery = oidc_discovery.new(self.http_backend)
```

Comment:
Isn't `self.http_backend` always `nil` at this point?

Reply:
Yes, unless you set it on the module level. `self` already has the metatable pointing to _M.

Reply:
:+1: